### PR TITLE
Export C API symbols in _pywrap_tensorflow_internal.so

### PR DESCRIPTION
**Pull request by fritzo**
This PR exports C API symbols so that python extension libraries can use the C API.

As noted in #7541, there is currently a conflict between `_pywrap_tensorflow_internal.so` and `libtensorflow.so`. This conflict prevents use of the tensorflow C API and the python API in the same process. This PR attempts to implement a workaround suggested by @jhseu by exporting the C API symbols in the python library, so that the Python and C APIs are both available from the single `_pywrap_tensorflow_internal.so`.

### CODE REVIEW

File: tensorflow/tf_exported_symbols.lds [hunk 1]
```diff
@@ -1,3 +1,4 @@
 *tensorflow*
 *perftools*gputools*
 *tf_*
+TF_*
```
Comment (jhseu) [inline tensorflow/tf_exported_symbols.lds]: This shouldn't be necessary due to the `*tf_*` above.

File: tensorflow/tf_version_script.lds [hunk 2]
```diff
@@ -2,6 +2,7 @@ tensorflow {
   global:
     *tensorflow*;
     *perftools*gputools*;
+    TF_*;
```
Comment (jhseu) [inline tensorflow/tf_version_script.lds]: *tf_* instead?

Comment (fritzo) [inline tensorflow/tf_version_script.lds]: This appears to be case sensitive, and the C API functions are all prefixed with `TF_`.

Comment (fritzo) [inline tensorflow/tf_version_script.lds]: Actually, I don't know whether it's case sensitive. I'll look around for documentation.

Comment (fritzo) [inline tensorflow/tf_version_script.lds]: I confirmed that `--exclude_symbols_list` is case sensitive by replacing `TF_` with `*tf_*` and observing that the `TF_New*` symbols were present with `TF_` but absent with `*tf_*`.

File: tensorflow/tf_exported_symbols.lds (hunk 1 above)
Comment (fritzo) [inline tensorflow/tf_exported_symbols.lds]: `--exclude_symbols_list` is case sensitive; see comment below.

**Comment by jhseu** [comment]
Seems fine to me. Adding Asim to see if he has any objections.

**Review by asimshankar — APPROVED** [review]
Looks good to me, thanks!

**Comment by jhseu** [comment]
Jenkins, test this please

**Comment by fritzo** [comment]
Thanks for reviewing, @jhseu!

**Pull request merged**